gdx/src/com/badlogic/gdx/graphics/glutils/GLFrameBuffer.java

```diff
@@ -350,6 +401,71 @@ public void end (int x, int y, int width, int height) {
+
+	/** Transfer pixels from this frame buffer to the destination frame buffer. Usually used when using multisample, it resolves
+	 * samples from this multisample FBO to a non-multisample as destination in order to be used as textures. This is a convenient
+	 * method that automatically choose which of stencil, depth, and colors buffers attachment to be copied.
+	 * @param destination the destination of the copy. */
+	public void transfer (GLFrameBuffer<T> destination) {
+
+		int copyBits = 0;
+		for (FrameBufferTextureAttachmentSpec attachment : destination.bufferBuilder.textureAttachmentSpecs) {
+			if (attachment.isDepth && bufferBuilder.hasDepthRenderBuffer || bufferBuilder.hasPackedStencilDepthRenderBuffer) {
+				copyBits |= GL20.GL_DEPTH_BUFFER_BIT;
+			} else if (attachment.isStencil && bufferBuilder.hasStencilRenderBuffer
+				|| bufferBuilder.hasPackedStencilDepthRenderBuffer) {
+				copyBits |= GL20.GL_STENCIL_BUFFER_BIT;
+			} else if (colorBufferHandles.size > 0) {
+				copyBits |= GL20.GL_COLOR_BUFFER_BIT;
+			}
+		}
+
+		transfer(destination, copyBits);
+	}
+
+	/** Transfer pixels from this frame buffer to the destination frame buffer. Usually used when using multisample, it resolves
+	 * samples from this multisample FBO to a non-multisample as destination in order to be used as textures.
+	 * @param destination the destination of the copy.
+	 * @param copyBits combination of GL20.GL_COLOR_BUFFER_BIT, GL20.GL_STENCIL_BUFFER_BIT, and GL20.GL_DEPTH_BUFFER_BIT. When
+	 *           GL20.GL_COLOR_BUFFER_BIT is present, every color buffers will be copied to each corresponding color texture
+	 *           buffers in the destination framebuffer. */
+	public void transfer (GLFrameBuffer<T> destination, int copyBits) {
+		Gdx.gl.glBindFramebuffer(GL30.GL_READ_FRAMEBUFFER, framebufferHandle);
+		Gdx.gl.glBindFramebuffer(GL30.GL_DRAW_FRAMEBUFFER, destination.framebufferHandle);
+
+		int colorBufferIndex = 0;
+		int attachmentIndex = 0;
+		for (FrameBufferTextureAttachmentSpec attachment : destination.bufferBuilder.textureAttachmentSpecs) {
+			if (attachment.isColorTexture()) {
+				Gdx.gl30.glReadBuffer(GL30.GL_COLOR_ATTACHMENT0 + colorBufferIndex);
+
+				singleInt.clear();
+				singleInt.put(GL30.GL_COLOR_ATTACHMENT0 + attachmentIndex);
+				singleInt.flip();
+				Gdx.gl30.glDrawBuffers(1, singleInt);
+
+				Gdx.gl30.glBlitFramebuffer(0, 0, getWidth(), getHeight(), 0, 0, destination.getWidth(), destination.getHeight(),
+					copyBits, GL20.GL_NEAREST);
+
+				copyBits = GL20.GL_COLOR_BUFFER_BIT;
+				colorBufferIndex++;
+			}
+			attachmentIndex++;
+		}
+		// case of depth and/or stencil only
+		if (copyBits != GL20.GL_COLOR_BUFFER_BIT) {
+			Gdx.gl30.glBlitFramebuffer(0, 0, getWidth(), getHeight(), 0, 0, destination.getWidth(), destination.getHeight(),
+				copyBits, GL20.GL_NEAREST);
+		}
+
+		// restore draw buffers for destination
+		Gdx.gl30.glDrawBuffers(destination.defaultDrawBuffers.limit(), destination.defaultDrawBuffers);
```

Comment:
When testing out the FBO transfer method, I was getting a NPE crash if transferring to a regular FBO that is not MRT.  `defaultDrawBuffers` is only available on FBO that has MRT or samples enabled. 

Locally I added a null check like so, and the transfer now works for non-MRT destination FBOS

```
if (destination.defaultDrawBuffers != null) {
	// restore draw buffers for destination
	Gdx.gl30.glDrawBuffers(destination.defaultDrawBuffers.limit(), destination.defaultDrawBuffers);
}
```

Reply:
indeed, i updated GL31FrameBufferMultisampleTest to use the real implementation and confirmed the issue. It's fixed now.